x/auction/keeper/msg_server_reveal_bid.go

```diff
@@ -0,0 +1,77 @@
+package keeper
+
+import (
+	"context"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/tendermint/tendermint/crypto"
+
+	"github.com/Stride-Labs/stride/v5/x/auction/types"
+)
+
+// Bidders send in the bid object with the original salt used to encrypt their sealed bid from earlier messages
+// If their sealed bid hash matches the hash we generate then nothing has changed and their revealed bid is entered
+
+func (k msgServer) RevealBid(goCtx context.Context, msg *types.MsgRevealBid) (*types.MsgRevealBidResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	ctx.Logger().Info(fmt.Sprintf("[auction]"))
+	ctx.Logger().Info(fmt.Sprintf("[auction] Reveal request for %+v", msg))
+
+	auctionPools := k.Keeper.GetAllAuctionPool(ctx)
+	for _, pool := range auctionPools {
+		if pool.GetId() == msg.GetPoolID() {
+			auction := pool.GetLatestAuction()
+			if auction != nil && auction.GetAlgorithm() == types.AuctionType_SEALEDBID {
+				sealedAuction := auction.GetAuction().(*types.SealedBidAuction)
+				if sealedAuction != nil && sealedAuction.GetStatus() == types.AuctionState_REVEAL {
+
+					creator := msg.GetCreator()
+					sealedBids := sealedAuction.GetSealedBids()
+					ctx.Logger().Info(fmt.Sprintf("[auction] Here are the sealedAuction %+v", pool.GetLatestAuction().GetSealedBidAuction()))
+					ctx.Logger().Info(fmt.Sprintf("[auction] Here are the sealedbids map %+v", sealedBids))
+
+					inSealedBids := false
+					sealedValue := ""
+					for _, sealedBid := range sealedBids {
```

Comment:
We need to avoid using for loop in development as much as possible.
I suggest to add id field in auction type and save a map info into kv store.
The format of the map key can be bytes(pool id+auction id+bidder address) and the value can be auction type string.

Reply:
+1, the main reason for this is efficiency. By introducing large loops like this, if we ever had lots of bids submitting for an auction, the gas cost to reveal a bid would become quite high. A Stride "block" only has so much gas, so this also introduces a maximum to the number of bids that could be submitted. A lookup up table is nice and constant time!